File: 733-flood-fill/733-flood-fill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        rows = len(image)
        cols = len(image[0])
        cur_color = image[sr][sc]
        
        # if we have already colored return
        if newColor == cur_color:
            return image
        
        # else perform dfs
        def dfs(r, c):
            if image[r][c] == cur_color:
                image[r][c] = newColor
                if r>=1: dfs(r-1,c)
                if r+1<rows: dfs(r+1, c)
                if c>=1: dfs(r, c-1)
                if c+1 < cols: dfs(r, c+1)
        
        dfs(sr, sc)
        return image

File: 733-flood-fill/733-flood-fill.ts
function floodFill(image: number[][], sr: number, sc: number, newColor: number): number[][] {
    const rows = image.length;
    const cols = image[0].length;
    const curColor =  image[sr][sc];
    if (curColor === newColor) return image;
    
    const dfs = (r: number,c: number) => {
      if(image[r][c] == curColor){
        image[r][c] = newColor;
        if(r>=1) dfs(r-1, c);
        if(r+1<rows) dfs(r+1,c);
        if(c>=1) dfs(r, c-1);
        if(c+1<cols) dfs(r,c+1)
      }
    }
    
    dfs(sr, sc);
    return image;
};